Environment/Environment.tex

```diff
@@ -1282,20 +1282,106 @@ \section{Other interesting tools}
+	[World imageForm] timeProfile.
+	sem signal] fork.
+	sem wait.
+	window := helper foregroundWindow.
+	helper scaleWindow: window extent: 580 px @ 425 px.
+	splitter := window allMorphs detect: [:m |
+	(m isKindOf: ProportionalSplitterMorph) and: [m splitsTopAndBottom]].
+	splitter repositionBy: 0 px @ 110 px.
+	helper selectAnyListIn: window at: '##Leaves##' andScrollIntoViewBy: 3.
+	list := helper listMorphIncludingMatch: '##Leaves##' in: window.
+	helper
+		type: '>>frameAndFillRectangle:fillStyle:borderStyle:' into: list;
+		keyStroke: list key: Character return.
+]
+\end{ExecuteSmalltalkScript}
+\begin{ExecuteSmalltalkScript}
+SBEScreenshotRecorder writeTo: './figures/SpaceTally.png' building: [:helper |
+	(SpaceTally new depth: 2; spaceTally: (PackageInfo named: #Morphic) classes) exploreWithLabel: 'SpaceTally'.
+	helper scaleWindow: helper foregroundWindow extent: 500 px @ 225 px.
+]
+\end{ExecuteSmalltalkScript}
+
+First, you can send the message \mthind{BlockClosure}{timeToRun} to a block to measure the time it takes to execute it:
+\begin{code}{NB: CANNOT TEST}
+[10000 factorial] timeToRun --> 27
+\end{code}
+\noindent
+Alternatively, you can select a piece of code in a workspace, debugger, \autc, open the yellow button menu, and select \menu{tally it}.
+If a single execution of the code is too fast to measure (confidently), you can also \emph{benchmark} it by sending \mthind{BlockClosure}{bench} or \mthind{BlockClosure}{benchFor:} to a block.
+This will run the block repeatedly and compute a few statistics about the execution time.
+\begin{code}{NB: CANNOT TEST}
+[10 factorial] bench --> '14,500,000 per second. 68.8 nanoseconds per run. 0 % GC time.'
+\end{code}
+\index{benchmarks}
+
+\begin{figure}[btp]
+	\begin{minipage}[t]{0.48\textwidth}
+		\ifluluelse
+			{\includegraphics[width=\textwidth]{TimeProfile}}
+			{\includegraphics[scale=0.7]{TimeProfile}}
+		\label{fig:TimeProfile}
+	\end{minipage}
+	\hfill
+	\begin{minipage}[t]{0.48\textwidth}
+		\ifluluelse
+			{\includegraphics[width=\textwidth]{SpaceTally}}
+			{\includegraphics[scale=0.7]{exploreWithLabel: 'SpaceTally'}}
+		\label{fig:SpaceTally}
+	\end{minipage}
+	\caption{A time profiler (left) and a space tally (right).}
+\end{figure}
+
+If you have discovered that your code is too slow, you might wonder which parts of your code are responsible for the slowness.
+For that, \sq provides the \emphind{message tally} toool.
+You can start it to record the execution times for your code by selecting the code and choosing \menu{spy on it} from the context menu.
+Alternatively, you can let it observe all processes in the image from the menu item \menu{world main docking bar~\go{} Apps~\go{} Message Tally}.
+Note that the message tally is a sampling profiler, so for detailed information, you will have to let your code run for some seconds (e.g. by running it 1000 times if it only takes little time).

```

Comment:
vll: "little time" -> "short time"

Reply:
According to DeepL Write, Google, and Grammarly, "a little time" (but I missed the "a") is actually more popular than "short time". :)

```suggestion
Note that the message tally is a sampling profiler, so for detailed information, you will have to let your code run for some seconds (e.g. by running it 1000 times if it only takes a little time).
```